maitake/src/task.rs

```diff
@@ -447,12 +446,32 @@ impl Header {
             Poll::Pending
         }
 
+        unsafe fn nop_deallocate(ptr: NonNull<Header>) {
+            unreachable!("stub task ({ptr:p}) should never be deallocated!");
+        }
+
         Self {
             run_queue: mpsc_queue::Links::new_stub(),
             state: StateCell::new(),
-            vtable: &Vtable { poll: nop },
+            vtable: &Vtable {
+                poll: nop,
+                deallocate: nop_deallocate,
+            },
         }
     }
+
+    unsafe fn drop_slow(this: NonNull<Self>) {
```

Comment:
the idea was that the ref-count decrement should be inlined, but the actual dealloc shouldn't be. this could be renamed to just deallocate or something.